core/templates/dev/head/domain/topic_viewer/ReadOnlyTopicObjectFactory.ts

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of frontend topic-data domain
+ * objects.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { Injectable } from '@angular/core';
+
+import { Subtopic, SubtopicObjectFactory } from
+  'domain/topic/SubtopicObjectFactory';
+
+import { SkillSummary, SkillSummaryObjectFactory } from
+  'domain/skill/SkillSummaryObjectFactory';
+
+export class ReadOnlyTopic {
+  _topicName: string;
+  _topicId: string;
+  _canonicalStories: any;
+  _additionalStories: any;
+  _uncategorizedSkills: Array<SkillSummary>;
+  _subtopics: Array<Subtopic>;
+  _degreesOfMastery: any;
+  _skillDescriptions: any;
+
+  constructor(topicName: string, topicId: string, canonicalStories: any,
+      additionalStories: any, uncategorizedSkills: Array<SkillSummary>,
+      subtopics: Array<Subtopic>, degreesOfMastery: any, skillDescriptions) {
+    this._topicName = topicName;
+    this._topicId = topicId;
+    this._canonicalStories = canonicalStories;
+    this._additionalStories = additionalStories;
+    this._uncategorizedSkills = uncategorizedSkills;
+    this._subtopics = subtopics;
+    this._degreesOfMastery = degreesOfMastery;
+    this._skillDescriptions = skillDescriptions;
+  }
+
+  getTopicName(): string {
+    return this._topicName;
+  }
+
+  getTopicId(): string {
+    return this._topicId;
+  }
+
+  // Return type is any as there is no usage
```

Comment:
I didn't get it? If it's of no use then don't add it to the domain object.

Reply:
Right, I didn't convey it properly as per https://github.com/oppia/oppia/pull/8476#discussion_r374216609. Though I found `canonicalStoriesList` being used in `topic-viewer-page.directive.html. So, thought better keep it rather than causing error.